contracts/core/liquidators/utils/LiquidatorUtils.sol

```diff
@@ -0,0 +1,150 @@
+ */
+library LiquidatorUtils {
+    using SafeMath for uint256;
+    using CommonMath for uint256;
+
+    /* ============ Internal Functions ============ */
+
+    /**
+     * Calculate the rebalance volume as the difference in allocation percentages times market
+     * cap.
+     *
+     * @param _currentSet                       The Set to rebalance from
+     * @param _nextSet                          The Set to rebalance to
+     * @param _oracleWhiteList                  OracleWhiteList used for valuation
+     * @param _currentSetQuantity               Quantity of currentSet to rebalance
+     */
+    function calculateRebalanceVolume(
+        ISetToken _currentSet,
+        ISetToken _nextSet,
+        IOracleWhiteList _oracleWhiteList,
+        uint256 _currentSetQuantity
+    )
+        internal
+        view
+        returns (uint256)
+    {
+        // Calculate current set value
+        uint256 currentSetValue = SetUSDValuation.calculateSetTokenDollarValue(
+            _currentSet,
+            _oracleWhiteList
+        );
+
+        // Calculate allocationAsset's current set allocation
+        address allocationAsset = _currentSet.getComponents()[0];
+        uint256 currentSetAllocation = calculateAssetAllocation(
+            _currentSet,
+            _oracleWhiteList,
+            allocationAsset
+        );
+
+        // Calculate allocationAsset's next set allocation
+        uint256 nextSetAllocation = calculateAssetAllocation(
+            _nextSet,
+            _oracleWhiteList,
+            allocationAsset
+        );
+
+        // Get allocation change
+        uint256 allocationChange = currentSetAllocation > nextSetAllocation ?
+            currentSetAllocation.sub(nextSetAllocation) :
+            nextSetAllocation.sub(currentSetAllocation);
+
+        // Return rebalance volume by multiplying allocationChange by Set market cap, deScaling to avoid
+        // overflow potential, still have 18 decimals of precision
+        return currentSetValue.mul(_currentSetQuantity).deScale().mul(allocationChange).deScale();
+    }
+
+    /**
+     * Calculate the allocation percentage of passed asset in Set
+     *
+     * @param _setToken             Set being evaluated
+     * @param _oracleWhiteList      OracleWhiteList used for valuation
+     * @param _asset                Asset that's allocation being calculated
+     */
+    function calculateAssetAllocation(
+        ISetToken _setToken,
+        IOracleWhiteList _oracleWhiteList,
+        address _asset
+    )
+        internal
+        view
+        returns (uint256)
+    {
+        address[] memory components = _setToken.getComponents();
+
+        // Get index of asset and return if asset in Set
+        (
+            uint256 assetIndex,
+            bool isInSet
+        ) = AddressArrayUtils.indexOf(components, _asset);
+
+        // Calculate allocation of asset or return 0 if not in Set
```

Comment:
Why can't we just call `calculateTokenAllocationAmountUSD` and divided by `calculateSetTokenDollarValue`?

Reply:
Less efficient

Reply:
This way we only have to calculate component value twice instead of three times